File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.cpp
class Solution {
public:
    int kthSmallest(vector<vector<int>>& matrix, int k) {
        int n = matrix.size();
        int m = matrix[0].size();

        vector<int> val;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                val.push_back(matrix[i][j]);
            }
        }
        sort(val.begin(), val.end());

        return val[k - 1];
    }
};

File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.py
class Solution(object):
    def kthSmallest(self, matrix, k):

        n = len(matrix)

        left = matrix[0][0]
        right = matrix[n - 1][n - 1]

        while left < right:

            mid = left + (right - left) // 2

            count = 0
            row = n - 1
            col = 0

            while row >= 0 and col < n:

                if matrix[row][col] <= mid:
                    count += row + 1
                    col += 1
                else:
                    row -= 1

            if count >= k:
                right = mid
            else:
                left = mid + 1

        return left